### PR TITLE
deps: new structure and update versions

### PR DESCRIPTION
**Pull request by Rqnsom**
- new toml structure takes advantage of using workspace dependencies
- all specified dependencies are updated to the latest version

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,9 +15,45 @@ edition = "2021"
 homepage = "https://aptoslabs.com"
```
Comment (asmie) [inline Cargo.toml:15]: Shouldn't we change that too?

**Review by asmie — CHANGES_REQUESTED** [review]
Let's change the homepage too 😸

File: Cargo.toml (hunk 1 above)
Comment (Rqnsom) [reply Cargo.toml]: Done (after quick discussion on Slack) 😄

**Review by asmie — APPROVED** [review]
👍

**Pull request merged**